services/pkgreview/package-rating.service.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { starRating, metric } = require('../text-formatters')
+const { colorScale } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const pkgReviewColor = colorScale([2, 3, 4])
+
+const schema = Joi.object({
+  rating: Joi.number()
+    .min(0)
+    .max(1)
+    .precision(1)
+    .required()
+    .allow(null),
+  reviewsCount: nonNegativeInteger,
+}).required()
+
+module.exports = class PkgreviewRating extends BaseJsonService {
+  static get category() {
+    return 'rating'
+  }
+
+  static get route() {
+    return {
+      base: 'pkgreview',
+      pattern: ':format(rating|stars)/:pkgManager/:pkgSlug+',
```

Comment:
Since the acceptable values for `pgkManager` are constrained by the upstream service (pkgreview), we can tighten up this route to avoid Shields unnecessarily accepting bogus values and making API calls to pkgreview.

Below suggestion only includes `npm` as the allowed value, though if you'd prefer to go ahead and include the other package managers you're planning to support that's fine too

```suggestion
      pattern: ':format(rating|stars)/:pkgManager(npm)/:pkgSlug+',
```

Reply:
Cool! Will do!

Reply:
Done!